kolena/io.py

```diff
@@ -42,7 +42,7 @@ def _deserialize_dataobject(x: Any) -> Any:
         return [_deserialize_dataobject(item) for item in x]
 
     if isinstance(x, dict):
-        if data_type := x.get(DATA_TYPE_FIELD):
+        if data_type := x.pop(DATA_TYPE_FIELD, None):
```

Comment:
I realized this changed prior behaviour when looking back at https://github.com/kolenaIO/kolena/pull/543/files#diff-903d27b9445c08b2a9562ef9d804bc4c5f3ff9e06054752dc952144c7da2b000L45